main_app/igdb_api.py

```diff
@@ -93,34 +100,82 @@ def api_get_last_pages_amount(self) -> int:
+        if amount > 200:
+            for i in range(4):
+                query = f'{category}/?fields={fields}&limit=50&offset={i * 50}'
+                result.extend(self.__api_get(query))
+            for i in range(round((amount - 200) / 10)):
+                games_list = [i * 10 + j for j in range(10)]
+                print(games_list)
+                query = f'{category}/{games_list}?fields={fields}'
+                result.extend(self.__api_get(query))
+        else:
+            short_val = math.floor(amount / 50)
+            for i in range(short_val):
+                query = f'{category}/?fields={fields}&limit=50&offset={i * 50}'
+                result.extend(self.__api_get(query))
+            if amount - short_val * 50 != 0:
+                query = f'{category}/?fields={fields}&limit={amount - short_val * 50}&offset={short_val * 50}'
+                result.extend(self.__api_get(query))
+        return result
+
+    def api_get_games_list(self, page: str = '1', game_ids: list = None, genres: bool = False, **kwargs) -> list:
         """kwargs:
         search: str = None,
         genres: str = None,
         platforms: str = None,
         user_ratings: tuple = None"""
         category = 'games/?'
         additional = {
-            'fields': 'name,cover,version_title,rating',
+            'fields': 'name,cover,version_title,rating,genres',
             'limit': self.__limit,
-            'offset': (int(page) - 1) * self.__limit}
-        if not kwargs.get('search'):
-            additional['order'] = 'popularity:desc'
-        filters = self.__generate_filters(**kwargs)
-        additional.update(filters)
+            'offset': (int(page) - 1) * self.__limit
+        }
+        if game_ids:
+            games = ','.join(map(str, game_ids))
+            category = f'games/{games}?'
+        else:
+            if not kwargs.get('search'):
+                additional['order'] = 'popularity:desc'
+            filters = self.__generate_filters(**kwargs)
+            additional.update(filters)
         encoded_url = urlencode(additional, quote_via=quote_plus)
         data = self.__api_get(f'{category}{encoded_url}')
         images_query = list(set(each.get('cover', '') for each in data if each.get('cover')))
-        images = dict(self.api_get_image(images_query, True))
+        images = self.api_get_image(images_query, True)
+        genres_list = []
         for i, each in enumerate(data):
             if each:
                 if each.get('cover'):
                     data[i]['cover'] = images.get(each.get('cover'))
+                if genres:
+                    genres_list.append(data[i].get('genres', [0])[0])
+        if genres:
+            j = 0
+            genre_names = self.api_get_names('genres', genres_list)
+            print(genre_names)
+            for i, each in enumerate(data):
+                if each.get('genres'):
+                    data[i]['genres'] = genre_names[each.get('genres')[0]]
```

Comment:
`data[i]['genres'] = genre_names[each['genres'][0]]`
Please do not overuse python `.get`, Always try accessing dict by key first
If you will get `None` after `.get('genres')` there will be error due to accessing like `None[0]`